cpp/src/randomforest/randomforest.cu

```diff
@@ -341,6 +344,120 @@ std::vector<unsigned char> save_model(ModelHandle model) {
   return bytes_info;
 }
 
+void check_concat_tl_mod(ModelHandle concat_tree_handle,
+                         std::vector<ModelHandle> treelite_handles,
+                         bool deep_check) {
+  size_t concat_forest;
+  size_t single_handle;
+  int total_num_trees;
+  for (int tl_num = 0; tl_num < treelite_handles.size(); tl_num++) {
+    TREELITE_CHECK(
+      TreeliteQueryNumTree(treelite_handles[tl_num], &single_handle));
+    total_num_trees = total_num_trees + single_handle;
+  }
+
+  TREELITE_CHECK(TreeliteQueryNumTree(concat_tree_handle, &concat_forest));
+
+  ASSERT(
+    concat_forest != total_num_trees,
+    "Error! the number of trees in the concatenated forest and the sum "
+    "of the trees present in the forests present in each worker are not equal");
+
+  size_t concat_feats, indiv_feats, concat_output, indiv_output;
+  TREELITE_CHECK(TreeliteQueryNumFeature(concat_tree_handle, &concat_feats));
+  TREELITE_CHECK(TreeliteQueryNumFeature(concat_tree_handle, &concat_output));
+
+  TREELITE_CHECK(TreeliteQueryNumFeature(treelite_handles[0], &indiv_feats));
+  TREELITE_CHECK(
+    TreeliteQueryNumOutputGroups(treelite_handles[0], &indiv_output));
+
+  ASSERT(concat_feats != indiv_feats || concat_output != indiv_output,
+         "Error! the number of features or the output group in the "
+         "concatenated forest "
+         "and the forests present in each worker are not equal");
+
+  if (deep_check) {
```

Comment:
`if (!deep_check) return;`. In this case, you don't need extra indent for the subsequent code.